keras/utils/data_utils.py

```diff
@@ -654,7 +654,7 @@ def get(self):
                     yield inputs
             else:
                 all_finished = all([not thread.is_alive() for thread in self._threads])
-                if all_finished:
```

Comment:
Between the validating that the queue is empty and checking that all threads are inactive, the final running thread might update the queue. The conditions for this to happen must be just right and it's difficult to reproduce. By placing sleep() calls in strategic places one can reproduce the bug.